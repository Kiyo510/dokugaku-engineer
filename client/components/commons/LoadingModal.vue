<template>
  <div>
    <transition name="fade">
      <div v-if="showModal" class="modal">
        <div class="loading">
          <i class="fad fa-spinner fa-spin fa-lg" />
        </div>
      </div>
    </transition>
  </div>
</template>

<style lang="scss" scoped>
.modal {
  background-color: rgba(0, 0, 0, 0.6);
  height: 100vh;
  left: 0;
  overflow: hidden;
  position: fixed;
  top: 0;
  width: 100vw;
  z-index: 2000;
}

.loading {
  color: $color-white1;
  left: 47%;
  position: absolute;
  top: 40%;
}

.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.1s ease-out;
}

.fade-enter,
.fade-leave-to {
  opacity: 0;
}
</style>

<script>
export default {
  props: {
    showModal: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
