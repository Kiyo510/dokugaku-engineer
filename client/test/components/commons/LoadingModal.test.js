import { mount } from '@vue/test-utils'
import LoadingModal from '~/components/commons/LoadingModal.vue'

describe('components/commons/LoadingModal.vue', () => {
  test('match snapshot', () => {
    const wrapper = mount(LoadingModal)
    expect(wrapper.element).toMatchSnapshot()
  })
})
