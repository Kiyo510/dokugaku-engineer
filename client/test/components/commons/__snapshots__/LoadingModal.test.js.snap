// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/LoadingModal.vue match snapshot 1`] = `
<div>
  <transition-stub
    name="fade"
  >
    <!---->
  </transition-stub>
</div>
`;
